core/service.go

```diff
@@ -586,10 +596,10 @@ func (svc *Service) startTunnel(ctx context.Context, bk *buildkit.Client, svcs *
 		},
 		Host:  dialHost,
 		Ports: ports,
-		Stop: func(context.Context) error {
+		Stop: func(_ context.Context, force bool) error {
 			stop()
 			// HACK(vito): do this async to prevent deadlock (this is called in Detach)
-			go svcs.Detach(svcCtx, upstream)
+			go svcs.Detach(svcCtx, upstream, nil, force)
```

Comment:
So am I correct that this means if you start a service via a host tunnel, then the "actual" service container is always SIGKILL'd right now?

If so, maybe we could just set the default 10s grace period here too rather than `nil` so that the codepaths where it's still gracefully stopped if `force==false` get activated.

There's still a separate problem where `Stop` won't block until the service container has actually stopped. That gets into much more murky territory so don't mind if we just leave this behavior as is in this PR though.

Reply:
> So am I correct that this means if you start a service via a host tunnel, then the "actual" service container is always SIGKILL'd right now?

Ok, so if it's started as part of this, then calling `stop` would apply whatever `kill` was applied to the `stop` call.

Here's that scenario:
- I create a service
- I create a tunnel for that service and start it
- I stop the tunnel - the tunnel is stopped, and the service is stopped (applying `kill` if set)

Here's a slightly different case (not sure if this actually happens tbf, need some tests on this):
- I create a service and start it
- I create a tunnel for that service and start it
- I stop the tunnel - the tunnel is not stopped, it's detached - and we immediately return
- I stop the service - the service is stopped (applying `kill` if set)

What's less apparent to me is what should happen if:
- I create a service
- I connect it to a container
- I create a tunnel for that service and start it
- I stop the tunnel - the tunnel is not stopped, it's detached since the container is still running - and we immediately return
- The container stops, and the service is gracefully killed
  - This seems really weird to me - the user made a request for the underlying service to be stopped with a set of options, but those weren't really applied, instead we gracefully exit.